src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -70,6 +70,9 @@ export class FileStat implements IFileStat {
 	public hasChildren: boolean;
 	public children: FileStat[];
 	public parent: FileStat;
+	public isVirtualDirectory: boolean;
+	public isVirtualDirectoryMember: boolean;
+	public virtualDirectoryName: string;
```

Comment:
Doesn't it need tests?

Reply:
Yes, this should have some tests. Additionally, how does this perform with large file collections?